lib/sync/remote-resin-io-device.coffee

```diff
@@ -166,33 +188,41 @@ exports.sync = ({ uuid, baseDir, destination, before, after, ignore, port = 22,
 			startMessage: "Syncing to #{destination} on #{uuid.substring(0, 7)}..."
 			stopMessage: "Synced #{destination} on #{uuid.substring(0, 7)}."
 
-	Promise.props
-		fullUuid: getDeviceInfo(uuid)
-		username: resin.auth.whoami()
+	Promise.join(
+		getDeviceInfo(uuid)
+		resin.auth.whoami()
+		({ fullUuid, isAUFS }, username) ->
+			return { fullUuid, isAUFS, username }
+	)
 	.tap -> # run 'before' action
 		if before?
 			shell.runCommand(before, baseDir)
-	.then ({ fullUuid, username }) -> # stop container
-		stopContainerSpinner(resin.models.device.stopApplication(uuid)).then (containerId) ->
-			# the resolved 'containerId' value is needed for the rsync process over resin-proxy
-			return { containerId, fullUuid, username }
-	.then ({ containerId, fullUuid, username }) -> # sync container
-		syncContainer({ fullUuid, username, containerId, baseDir, destination })
-		.then -> # start container
-			startContainerSpinner(resin.models.device.startApplication(uuid))
-		.then -> # run 'after' action
-			if after?
-				shell.runCommand(after, baseDir)
-		.then ->
-			console.log(chalk.green.bold('\nresin sync completed successfully!'))
-		.catch (err) ->
-			# Notify the user of the error and run 'startApplication()'
-			# once again to make sure that a new app container will be started
-			startContainerAfterErrorSpinner(resin.models.device.startApplication(uuid))
+	.then ({ fullUuid, isAUFS , username }) -> # get container id
+		# the resolved 'containerId' value is needed for the rsync process over resin-proxy
+		infoContainerSpinner(resin.models.device.getApplicationInfo(uuid))
+		.tap ->
+			if not isAUFS # for AUFS devices, rsync has to be performed while the container is running
```

Comment:
Yeah that sounds right, will do.

So if you `resin-sync` incorrect code, and a subsequent `resin-sync` does not solve the issue, the recovery action is, as you said, to simply restart the app on the device, which will recreate the container and reset its status